sync_merkleroots_test.go

```diff
@@ -0,0 +1,104 @@
+package walletclient
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/fixtures"
+	"github.com/bitcoin-sv/spv-wallet-go-client/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSyncMerkleRoots(t *testing.T) {
+
+	t.Run("Should properly sync database when empty", func(t *testing.T) {
+		// setup
+		server := fixtures.MockMerkleRootsAPIResponseNormal()
+		defer server.Close()
+
+		// given
+		repo := fixtures.CreateRepository([]models.MerkleRoot{})
+		client, err := NewWithXPriv(server.URL, fixtures.XPrivString)
+		require.NotNil(t, client.xPriv)
+		require.NoError(t, err)
+
+		// when
+		err = client.SyncMerkleRoots(context.Background(), repo, 0)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, len(fixtures.MockedSPVWalletData), len(repo.MerkleRoots))
```

Comment:
consider using `require.Len` which will put more precise message in case of failure